tscv/_split.py

```diff
@@ -163,12 +163,17 @@ def __complement_masks(self, masks):
 
     def __complement_indices(self, indices, n_samples):
         before, after = self.gap_before, self.gap_after
+        allindices = np.arange(n_samples)
         for index in indices:
-            complement = np.arange(n_samples)
-            for i in index:
-                begin = max(i - before, 0)
-                end = min(i + after + 1, n_samples)
-                complement = np.setdiff1d(complement, np.arange(begin, end))
+            # split index in subarrays of contiguous elements
+            subindexes = np.split(index, np.where(np.diff(index) != 1)[0] + 1)
+            complement = allindices
+            # find complement on arrays of contiguous elements
+            for subindex in subindexes:
+                begin = max(0, subindex[0] - before)
+                end = min(subindex[-1] + after + 1, n_samples)
+                complement = np.intersect1d(complement,
+                    np.setdiff1d(allindices, allindices[begin:end]))
```

Comment:
You can simplify this line by using the fact $A \cap B^C = A \setminus  B$.

Reply:
Hi @WenjieZ , I'm sorry, I didn't get it, can you reformulate your statement?

Reply:
`complement = np.setdiff1d(complement, allindices[begin:end])`

![math](https://user-images.githubusercontent.com/6860682/152519945-68722733-f69b-4805-8c2f-673d037f9062.png)

Reply:
Ok, right! fixed!
Thank you